OEL-446: OE Showcase scaffolding

abel-santos-corral opened the pull request:
We need to create the scaffolding for oe_showcase. Add base elements

abel-santos-corral committed:
OEL-446: Adding basic scaffold fields.
Saving work in progress.
Adding README.md getting oe_profile as template.
Added first round of elements in composer.json.
Added all yml.dist or xml.dist files from template.

abel-santos-corral committed:
OEL-446: Change version of code-review and eliminate coder at composer.json component.

abel-santos-corral committed:
OEL-446: Eliminate activation of oe_showcase at runner.yml.dist component.

abel-santos-corral committed:
OEL-446: Fix phpunit version at composer.json.

abel-santos-corral committed:
OEL-446: Fix references of theme at README.md component.

abel-santos-corral committed:
OEL-446: Add selenium environment vars and removing php version 7.2.

abel-santos-corral committed:
OEL-446: Fix references of bartik theme.

abel-santos-corral committed:
OEL-446: Fix versions of packages and eliminate those not used at composer.json.

abel-santos-corral committed:
OEL-446: Add oe_showcase.profile component.

abel-santos-corral committed:
OEL-446: Add oe_showcase.info.yml component.

abel-santos-corral committed:
OEL-446: Fix the core version parameter at info.yml file.

drishu commented on README.md:
```diff
@@ -1 +1,109 @@
+
+## Rationale
+
+Opting for a minimalistic installation profile will make it easier to deal with a fully distributed approach: the
```
The rationale is different here, the showcase is for showcasing the openeuropa library features, so that developers have examples as guid to how to build these features. Try to word that idea better though.

drishu commented on README.md:
```diff
@@ -1 +1,109 @@
+To initiate step debugging from the browser, set the correct cookie using a browser extension or a bookmarklet
+like the ones generated at https://www.jetbrains.com/phpstorm/marklets/.
+
+[1]: https://www.drupal.org/project/bartik
```
remove this line, no need to put emphasis on it as we will change it to oe_whitelabel in near future

drishu commented on README.md:
```diff
@@ -1 +1,109 @@
+Basic installation profile, all it does is:
+
+1. Enable the bare minimum amount of core modules.
+2. Setup `seven` as administrative theme and [`bartik`][1] as front-end theme.
```
```suggestion
2. Setup `seven` as administrative theme.
```

we will change bartik in near future so lets not mention it

drishu commented on .drone.yml:
```diff
@@ -0,0 +1,102 @@
+      - NODE_MAX_INSTANCES=5
+      - NODE_MAX_SESSION=5
+
+  sparql:
```
this isn't used at the moment, lets remove it

drishu commented on docker-compose.yml:
```diff
@@ -0,0 +1,62 @@
+      MYSQL_ALLOW_EMPTY_PASSWORD: "yes"
+    # ports:
+    #   - 3306:3306
+  sparql:
```
this isn't used at the moment, lets remove it

drishu commented on runner.yml.dist:
```diff
@@ -0,0 +1,66 @@
+        - "vendor"
+        - "${drupal.root}"
+      file_private_path: "sites/default/files/private"
+    databases:
```
this isn't used at the moment, lets remove it

drishu commented on docker-compose.yml:
```diff
@@ -0,0 +1,62 @@
+      - .:/var/www/html # Non Mac users.
+      # - nfsmount:/var/www/html # Mac Users with the nfsmount volume.
+    environment:
+      # XDEBUG_CONFIG: "client_host=172.17.0.1" # Non-Mac users.
```
uncomment this

drishu commented on docker-compose.yml:
```diff
@@ -0,0 +1,62 @@
+    environment:
+      # XDEBUG_CONFIG: "client_host=172.17.0.1" # Non-Mac users.
+      # XDEBUG_CONFIG: remote_host=host.docker.internal remote_port=9000 remote_enable=1, idekey=VSCODE
+      XDEBUG_CONFIG: "client_host=172.17.0.1 client_port=9000  remote_host=172.17.0.1 remote_port=9000 remote_enable=1, idekey=VSCODE"
```
dont add this, if you need it, add a docker-compose.override.yml (into your local install, not in the codebase)

drishu commented on docker-compose.yml:
```diff
@@ -0,0 +1,62 @@
+      # - nfsmount:/var/www/html # Mac Users with the nfsmount volume.
+    environment:
+      # XDEBUG_CONFIG: "client_host=172.17.0.1" # Non-Mac users.
+      # XDEBUG_CONFIG: remote_host=host.docker.internal remote_port=9000 remote_enable=1, idekey=VSCODE
```
```suggestion
      # XDEBUG_CONFIG: "client_host=host.docker.internal" # Mac users.
```

drishu commented on docker-compose.yml:
```diff
@@ -0,0 +1,62 @@
+      XDEBUG_MODE: "debug"
+      PHP_IDE_CONFIG: "serverName=Docker"
+      # Enable step debugging for all PHP request. See ./README.md#step-debugging for more information.
+      XDEBUG_SESSION: "VSCODE"
```
```suggestion
      # XDEBUG_SESSION: 1
```
again, add your own config in override

drishu commented on phpunit.xml.dist:
```diff
@@ -0,0 +1,17 @@
+    <env name="SIMPLETEST_IGNORE_DIRECTORIES" value="vendor,node_modules,${drupal.root}"/>
+    <env name="SIMPLETEST_BASE_URL" value="${drupal.base_url}"/>
+    <env name="SIMPLETEST_DB" value="mysql://${drupal.database.user}:${drupal.database.password}@${drupal.database.host}:${drupal.database.port}/${drupal.database.name}"/>
+    <env name="SIMPLETEST_SPARQL_DB" value="sparql://${drupal.sparql.host}:${drupal.sparql.port}/?module=sparql_entity_storage"/>
```
remove this for now

drishu commented on phpunit.xml.dist:
```diff
@@ -0,0 +1,17 @@
+    <env name="MINK_DRIVER_ARGS_WEBDRIVER" value='["${selenium.browser}", null, "${selenium.host}:${selenium.port}/wd/hub"]'/>
+  </php>
+  <testsuites>
+    <testsuite>
```
add name property to testsuite

drishu commented on runner.yml.dist:
```diff
@@ -0,0 +1,66 @@
+    name: "oe_showcase"
+    user: "root"
+    password: ""
+  sparql:
```
remove for now

drishu commented on runner.yml.dist:
```diff
@@ -0,0 +1,66 @@
+  setup:behat:
+    - { task: "process", source: "behat.yml.dist", destination: "behat.yml" }
+  # Use the following commands only if you can run Task Runner commands on your host machine.
+  sparql:purge:
```
remove all this sparql stuff till needed

abel-santos-corral committed:
OEL-446: Eliminate references of theming by now and fixing rationale from README.md.

abel-santos-corral committed:
OEL-446: ELiminate references to sparql in several components.

abel-santos-corral committed:
OEL-446: Fix profile var at runner component.

abel-santos-corral committed:
OEL-446: Change environment part at docker-compose.yml component.

abel-santos-corral committed:
OEL-446: Remove unneeded stuff, and add description to testsuite.

abel-santos-corral committed:
OEL-446: Remove selenium to set all as per oe_profile example.

abel-santos-corral commented:
Some extra changes were added:

- Removing minimal from profile variable at runner component
- Removed all stuff from components regarding the selenium part. It was not present in the oe_profile model we had. We brought it from another example (oe_authentication).
- Added two more instructions on runner to add the un-aggregation of CSS and JS. At testing we saw that content page was not properly shown. We tried to add 777 permissions to files folder, but that alone didn't work.

abel-santos-corral committed:
OEL-446: Add drush instructions to avoid CSS and JS aggregation upon installation.

abel-santos-corral committed:
OEL-446: Fix method name at oe_showcase.profile.

abel-santos-corral committed:
OEL-446: Changing symlink as we did per C9 compatibility changes at runner component.

abel-santos-corral committed:
OEL-446: Add an option at phpunit.xml.dist.

drishu committed:
OEL-446: Fix testing setup.

GilNovacomm reviewed (approved):
Review ok.

Maxfire reviewed (approved):
Changes reviewed, looks good, thanks
